app/views/layouts/application.html.erb

```diff
@@ -12,8 +12,16 @@
     <nav class = "topnav">
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", "/register" %>
+      <% if current_user %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Profile", "/profile" %>
+        <%= link_to "Logout", "/logout", method: :delete %>
+        <div>Logged in as <%= current_user.name %></div>
+      <% else %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
```

Comment:
This seems like the same as 16 so cant it be outside the conditional?

Reply:
I grabbed this from Melrob & Phil's user story 3 code, they have a reason for it; some user types won't be able to see the cart.

Reply:
alrighty